test/init_test.go

```diff
@@ -44,13 +44,13 @@ import (
 
 var initMetrics sync.Once
 
-func setup(t *testing.T, fn ...func(*testing.T, *clients, string)) (*clients, string) {
+func Setup(t *testing.T, fn ...func(*testing.T, *Client, string)) (*Client, string) {
 	t.Helper()
 	namespace := names.SimpleNameGenerator.RestrictLengthWithRandomSuffix("arendelle")
```

Comment:
This class seems very tied to the specific way that Tekton tests itself, and it doesn't seem well designed to be reused by other tests. At the same time it seems almost trivially easy to copy and maintain your own `Clients` type that does the same thing. Making this test utility available to other consumers might make it hard to make changes to it in the future. I think I'd prefer you to create your own `Clients` wrapper type, based on this, instead of reusing this one.